modules/transport/PostMessageTransportBackend.js

```diff
@@ -0,0 +1,163 @@
+import Postis from 'postis';
+
+/**
+ * The default options for postis.
+ *
+ * @type {Object}
+ */
+const DEFAULT_POSTIS_OPTIONS = {
+    window: window.opener || window.parent
+};
+
+/**
+ * The list of methods of incomming postis messages that we have to support for
+ * backward compatability for the users that are directly sending messages to
+ * Jitsi Meet (without using external_api.js)
+ *
+ * @type {string[]}
+ */
+const LEGACY_INCOMING_METHODS = [
+    'avatar-url',
+    'display-name',
+    'email',
+    'toggle-audio',
+    'toggle-chat',
+    'toggle-contact-list',
+    'toggle-film-strip',
+    'toggle-share-screen',
+    'toggle-video',
+    'video-hangup'
+];
+
+/**
+ * The list of methods of outgoing postis messages that we have to support for
+ * backward compatability for the users that are directly listening to the
+ * postis messages send by Jitsi Meet(without using external_api.js).
+ *
+ * @type {string[]}
+ */
+const LEGACY_OUTGOING_METHODS = [
+    'display-name-change',
+    'incoming-message',
+    'outgoing-message',
+    'participant-joined',
+    'participant-left',
+    'video-conference-joined',
+    'video-conference-left',
+    'video-ready-to-close'
+];
+
+/**
+ * The postis method used for all messages.
+ *
+ * @type {string}
+ */
+const POSTIS_METHOD_NAME = 'data';
+
+/**
+ * Implements message transport using the postMessage API.
+ */
+export default class PostMessageTransportBackend {
+    /**
+     * Creates new PostMessageTransportBackend instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor({ enableLegacyFormat, postisOptions } = {}) {
+        this.postis = Postis({
+            ...DEFAULT_POSTIS_OPTIONS,
+            ...postisOptions
+        });
+
+        this._enableLegacyFormat = enableLegacyFormat;
+
+        if (!this._enableLegacyFormat) {
+            // backward compatability
+            LEGACY_INCOMING_METHODS.forEach(method =>
+                this.postis.listen(
+                    method,
+                    params => this._legacyDataReceiveCallback(method, params)));
+        }
+
+        this._receiveCallback = () => {
+            // Do nothing until a callback is set by the consumer of
+            // PostMessageTransportBackend via setReceiveCallback.
+        };
+
+        this.postis.listen(
+            POSTIS_METHOD_NAME,
+            data => this._receiveCallback(data));
+    }
+
+    /**
+     * Handles incomming legacy postis data.
+     *
+     * @param {string} method - The method property from postis data object.
+     * @param {Any} params - The params property from postis data object.
+     * @returns {void}
+     */
+    _legacyDataReceiveCallback(method, params = {}) {
+        this._receiveCallback({
+            data: {
+                name: method,
+                data: params
```

Comment:
The use of the word data in multiple places throughout this PR confused me a lot while I was reading it. I did try to use function argument deconstructing at times to remove some `data` entities but I still don't feel comfortable enough. There's still `data` properties, `_sendLegacyData` which I'd rather call `_legacySend` (and respectively, `_legacyDataReceiveCallback` which I renamed but would rename further to `_legacyReceiveCallback`. Generally, I tried and would prefer that we reduce the use of the word data in order to improve readability (for me at least).